source/fundamentals/connection/mongoclientsettings.txt

```diff
@@ -0,0 +1,197 @@
+- To connect to ``MULTIPLE`` servers in your cluster
+- The server selection to take at most ``5 SECONDS``
+
+.. literalinclude:: /includes/fundamentals/code-snippets/mcs.java
+   :start-after: begin ClusterSettings
+   :end-before: end ClusterSettings
+   :language: java
+   :emphasize-lines: 3-6
+   :dedent:
+
+For a list of methods you can use to specify your cluster settings, see
+the `API Documentation <{+api+}/apidocs/mongodb-driver-core/com/mongodb/connection/ClusterSettings.Builder.html>`__.
+
+.. _mcs-socket-settings:
+
+Socket Settings
+---------------
+
+Use the `applyToSocketSettings() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/MongoClientSettings.Builder.html#applyToSocketSettings(com.mongodb.Block)>`__
+method to specify settings on how to connect to a MongoDB server.
+
+Example
+~~~~~~~
+
+This example specifies the following settings for connecting to a
+MongoDB socket:
+
+- The connect to the socket within ``1000 MILLISECONDS``
+- To read from the socket within ``10 SECONDS``
+
+.. literalinclude:: /includes/fundamentals/code-snippets/mcs.java
+   :start-after: begin SocketSettings
+   :end-before: end SocketSettings
+   :language: java
+   :emphasize-lines: 4-5
+   :dedent:
+
+For a list of methods you can use to specify your socket settings, see
+the `API Documentation <{+api+}/apidocs/mongodb-driver-core/com/mongodb/connection/SocketSettings.Builder.html>`__.
+
+.. _mcs-connectionpool-settings:
+
+Connection Pool Settings
+------------------------
+
+Use the `applyToConnectionPoolSettings() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/MongoClientSettings.Builder.html#applyToConnectionPoolSettings(com.mongodb.Block)>`__
+method to specify settings related to a pool of connections to a MongoDB server.
+
+Example
+~~~~~~~
+
+This example specifies the following settings for connecting to a
+MongoDB connection pool:
+
+- The thread to wait at most ``5 SECONDS`` for an available connection
+- To have at most one connection associated with the pool
+- The pooled connection to live for at most ``100 MILLISECONDS``
+
+.. literalinclude:: /includes/fundamentals/code-snippets/mcs.java
+   :start-after: begin ConnectionPoolSettings
+   :end-before: end ConnectionPoolSettings
+   :language: java
+   :emphasize-lines: 4-6
+   :dedent:
+
+For a list of methods you can use to specify your connection pool
+settings, see the `API Documentation <{+api+}/apidocs/mongodb-driver-core/com/mongodb/connection/ConnectionPoolSettings.Builder.html>`__.
+
+.. _mcs-server-settings:
+
+Server Settings
+---------------
+
+Use the `applyToServerSettings() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/MongoClientSettings.Builder.html#applyToServerSettings(com.mongodb.Block)>`__
+method to specify settings on how to monitor each server.
+
+Example
+~~~~~~~
+
+This example specifies the following settings for connecting to a
+MongoDB server:
+
+- The heartbeat frequency to be at least ``10 MILLISECONDS``
```

Comment:
question: What would happen here if a primary is elected and its average ping is 12 ms?